packages/source-iottwinmaker/src/utils/propertyValueUtils.ts

```diff
@@ -0,0 +1,62 @@
+import { Type } from '@aws-sdk/client-iottwinmaker';
+import { isEmpty, isNil, isNumber, isString } from 'lodash';
+import { DATA_TYPE } from '@iot-app-kit/core';
+import type { DataValue, DataType as TMDataType } from '@aws-sdk/client-iottwinmaker';
+import type { Primitive, DataType } from '@iot-app-kit/core';
+
+/**
+ * Check if value is not null and not undefined.
+ *
+ * @param value The value to check.
+ * @returns Returns false if the value is nullish, otherwise true.
+ */
+export const isDefined = <T>(value: T | null | undefined): value is T => !isNil(value);
+
+/**
+ * Extracts the value out of a TwinMaker property's DataValue
+ *
+ * NOTE: Currently we treat booleans and objects as strings.
+ */
+export const toValue = (dataValue: DataValue): Primitive | undefined => {
+  const values = Object.values(dataValue).filter(isDefined);
+
+  if (isEmpty(values)) {
+    console.warn('Expected value to have at least one property value, but instead it has none!');
+    return undefined;
+  }
+
+  if (values.length > 1) {
+    console.warn('More than one value found in DataValue, use a random value only');
+  }
+
+  const value = values[0];
+
+  if (isNumber(value) || isString(value)) {
+    return value;
+  }
+
+  // Non-primitive value converts to string by default
+  return JSON.stringify(value);
+};
+
+/**
+ * Convert the TwinMaker DataType into AppKit DataType
+ *
+ * @param tmDataType the TwinMaker DataType to be converted.
+ * @returns the converted AppKit DataType
+ */
+export const toDataType = (tmDataType: TMDataType): DataType | undefined => {
```

Comment:
I remember when I was working on Athena data connector, if customer created a database using List/Map( advance  data type), Console was throwing an error. How do we want to handle here - https://docs.aws.amazon.com/iot-twinmaker/latest/apireference/API_DataType.html

Reply:
AppKit data type doesn't support list or map currently, so I convert them to strings for now. when we have the use case to handle list/map/table, we can update here to return them in proper format

Reply:
Why do we want to convert to string instead of saying not supported types? You can exclude them when you are reading the property and its data type?

Reply:
they can still be displayed in say overlay as string.